Get build passing in travis & begin moving documentation to roxygen format

georgestagg opened the pull request:
In the DESCRIPTION file `HenrikBengtsson/matrixStats@develop` has been added under `Remotes:`, so as to pick up a bugfix that lets this build without error in travis.

Presumably the develop branch is unstable and so this should be removed and `matrixStats (>= 0.50.XX)` added back into DESCRIPTION once a new version of matrixStats containing the changes that fix the problem hits CRAN.

georgestagg committed:
Use latest dev version of matrixStats to avoid a build bug

georgestagg committed:
Roxygenize functions and move data to correct loc

.rds files should live in ./inst/extdata

Re-add data as Rdata file and include documentation

roxygenize fitwafer

roxygenize fitall

roxygenize histplot

roxygenize fitmanova

roxygenize weekparam

roxygenize underparam

remove redundant demo

roxygenize curves

roxygenize plot

csgillespie commented:
Looks good. Two minor comments. For the roxygen tags, would you use `importFrom` instead of `import` to avoid namespace clashes. Also use `inheritParams` to avoid duplication.

georgestagg committed:
import->importFrom and fixed some typos

georgestagg committed:
Use inheritParams and slight tidy

georgestagg commented:
No problem, I've gone through and changed `import` to `importFrom` and used `inheritParams` where possible. 

I noticed during the change that nlme was imported but not used anywhere, perhaps something was changed in a previous commit and it was left behind?

I've removed nlme from DESCRIPTION for now, as Roxygen and Travis didn't seem to mind and `demo(worked_steps)` still works.

csgillespie commented on man/plotunder.Rd:
```diff
@@ -24,13 +24,14 @@ N.B. using the mean with low levels of data can lead to spurious curves,
 }
 \examples{
-wafers_folder = paste(path.package("voltagefit"),"/extdata/",sep="") # path to wafters data directory
+wafers_folder = paste(path.package("voltagefit"),"/extdata/",sep="") # path to wafers data directory
```
@extigy Would you change to `file.path` - platform independent.

csgillespie commented on man/undercurvesim.Rd:
```diff
@@ -20,7 +20,7 @@ A data.frame consisting of the fields:\cr
-wafers_folder = paste(path.package("voltagefit"),"/extdata/",sep="") # path to wafters data directory
+wafers_folder = paste(path.package("voltagefit"),"/extdata/",sep="") # path to wafers data directory
 fitted = fitall(wafers_folder)
 design = data.frame(week = c(1,1,1,1,2,2), wafer = unique(fitted$id), replicate = c(1:6), treatment = rep(1:6))
```
`c(1:6) = rep(1:6) = 1:6` @extigy

csgillespie commented on R/data_help_files.R:
```diff
@@ -0,0 +1,47 @@
+## Entry in data_help_files.R
```
@extigy The help file entries can be condensed into a single entry. I think it's the `aliases` tag?

csgillespie commented on R/model.R:
```diff
@@ -1,22 +1,30 @@
+#' @param iterlim   Maxiumum number of iterations to use in nlm (Default: 2000).
+#' @param id        WaferID (Default: A random string is generated as the ID).
+#' 
+#' @return A data.frame consisting of the fields:\cr
```
@extigy I don't think this will format correctly. You need to use a list.

csgillespie commented on R/model.R:
```diff
@@ -1,22 +1,30 @@
+#' @importFrom MASS mvrnorm
+#' 
+#' @export
 fitwafer = function(wafer, iterlim=2000, id=NULL){
```
@extigy Would you change all function names to have form: `fit_wafer`

georgestagg committed:
Fix documentation errors

georgestagg committed:
Rename and change some things around, for style

Functions renamed to abcd_efgh() format, and made verb-y where possible.

.R files renamed to abcd_efgh format.

Changed location of some curly braces for style+consistency.

Attempted to extend documentation description in places.

Fixed a couple typos.
